### PR TITLE
Remove `apm-server.sampling.keep_unsampled` config

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Remove `apm-server.sampling.keep_unsampled` config, and always drop unsampled transations. We'll follow up by retaining unsampled RUM transactions when the proposal in https://github.com/elastic/apm-server/issues/6260#issuecomment-955904292 is accepted.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
- [x] Documentation has been updated

## How to test these changes

Send some unsampled transactions to apm-server, check that they are not stored.

## Related issues

Closes https://github.com/elastic/apm-server/issues/5997

### CODE REVIEW

**Comment by axw** [comment]
System test failure is unrelated, and will be fixed by https://github.com/elastic/apm-server/pull/6525

File: sampling/sampling.go [hunk 1]
```diff
@@ -34,6 +34,8 @@ var (
 //
 // The returned model.BatchProcessor does not guarantee order preservation
 // of events retained in the batch.
+//
+// TODO(axw) only discard non-RUM unsampled transactions.
```
Comment (marclop) [inline sampling/sampling.go:38]: Follow up issue for this?

**Review by marclop — APPROVED** [review]

File: sampling/sampling.go (hunk 1 above)
Comment (axw) [reply sampling/sampling.go:38]: Yep, there's already an issue: https://github.com/elastic/apm-server/issues/6260

**Comment by marclop** [comment]
Verified with 9bb5e4cdd locally: Ran and verified that no transactions are stored.

```go
package main

import (
	"context"
	"fmt"
	"os"
	"time"

	"go.elastic.co/apm"
)

func main() {
	version := "undefined"
	if len(os.Args) > 1 {
		version = os.Args[1]
	}
	name := fmt.Sprintf("apm-server-%s", version)
	apm.DefaultTracer.SetSampler(apm.NewRatioSampler(0.0))
	tx := apm.DefaultTracer.StartTransaction(name, "type")
	ctx := apm.ContextWithTransaction(context.Background(), tx)
	span, ctx := apm.StartSpan(ctx, name, "type")
	span.Duration = time.Second
	span.End()
	tx.Duration = 2 * time.Second
	tx.End()
	apm.DefaultTracer.Flush(nil)
	fmt.Printf("%s: %+v\n", name, apm.DefaultTracer.Stats())
}
```

```console
$  go run main.go
apm-server-undefined: {Errors:{SetContext:0 SendStream:0} ErrorsSent:0 ErrorsDropped:0 TransactionsSent:1 TransactionsDropped:0 SpansSent:0 SpansDropped:0}
```

image

**Pull request merged**